internal/testrunner/runners/system/servicedeployer/compose.go

```diff
@@ -0,0 +1,177 @@
+package servicedeployer
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/common"
+	"github.com/elastic/elastic-package/internal/compose"
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/stack"
+)
+
+const (
+	stdoutFileName = "stdout"
+	stderrFileName = "stderr"
+)
+
+// DockerComposeRunner knows how to setup and teardown a service defined via
+// a docker-compose.yml file.
+type DockerComposeRunner struct {
+	ymlPath      string
+	project      string
+	stackNetwork string
+
+	stdout io.WriteCloser
+	stderr io.WriteCloser
+
+	stdoutFilePath string
+	stderrFilePath string
+}
+
+// NewDockerComposeRunner returns a new instance of a DockerComposeRunner.
+func NewDockerComposeRunner(ymlPath string) (*DockerComposeRunner, error) {
+	return &DockerComposeRunner{
+		ymlPath: ymlPath,
+	}, nil
+}
+
+// SetUp sets up the service and returns any relevant information.
+func (r *DockerComposeRunner) SetUp(ctxt common.MapStr) (common.MapStr, error) {
+	logger.Infof("setting up service using docker compose runner")
+	r.project = "elastic-package-service"
+
+	c, err := compose.NewProject(r.project, r.ymlPath)
+	if err != nil {
+		return ctxt, errors.Wrap(err, "could not create docker compose project for service")
+	}
+
+	// Boot up service
+	opts := compose.CommandOptions{
+		ExtraArgs: []string{"-d"},
+	}
+	if err := c.Up(opts); err != nil {
+		return ctxt, errors.Wrap(err, "could not boot up service using docker compose")
+	}
+
+	// Build service container name
+	serviceName, err := getStrFromCtxt(ctxt, "service.name")
+	if err != nil {
+		return ctxt, errors.Wrap(err, "could not get service name")
+	}
+	serviceContainer := fmt.Sprintf("%s_%s_1", r.project, serviceName)
+	ctxt.Put("Service.Hostname", serviceContainer)
+
+	// Redirect service container's STDOUT and STDERR streams to files in local logs folder
```

Comment:
I wonder if we can simplify this - we still have a volume with log files mounted, so maybe don't depend specifically on stdout and stderr, but just share a volume between agent and the service. I'm fine with discussing this in a different issue.

Reply:
I think I'm missing something here. Let's use the `apache` container as an example. Let's say we have a shared volume between this container and the `elastic-agent` container. I assume we want Apache log files to be placed by the `apache` container into this volume. But how/where exactly do we make this happen?

Reply:
I just read your overall PR review comment in which you mention this:

> In the service configuration, we define paths to be exposed as mounts...

I think that answers my question above, "how/where exactly do we make this happen?" We would define them in the `docker-compose.yml` file for the service. Let me try this out and see if we can make this work.